src/main/java/com/artipie/repo/ConfigFile.java

```diff
@@ -0,0 +1,115 @@
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.artipie.repo;
+
+import com.artipie.asto.Content;
+import com.artipie.asto.Key;
+import com.artipie.asto.Storage;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Supporting several config files extensions (e.g. `.yaml` and `.yml`).
+ *
+ * @since 0.14
+ */
+public final class ConfigFile {
+
+    /**
+     * Pattern to divide filename into two groups: name and extension.
+     */
+    private static final Pattern PTN = Pattern.compile("(?<name>.*)(\\.yaml|\\.yml)$");
+
+    /**
+     * Filename.
+     */
+    private final String filename;
+
+    /**
+     * Ctor.
+     * @param filename Filename
+     */
+    public ConfigFile(final String filename) {
+        this.filename = filename;
+    }
+
+    /**
+     * Ctor.
+     * @param filename Filename
+     */
+    public ConfigFile(final Key filename) {
+        this(filename.string());
+    }
+
+    /**
+     * Does file exist in the specified storage?
+     * @param storage Storage where the file with different extensions is checked for existence
+     * @return True if a file with either of the two extensions exists, false otherwise.
+     */
+    public CompletionStage<Boolean> existsIn(final Storage storage) {
+        final String name = this.trimExtension().orElse(this.filename);
+        final Key yaml = new Key.From(String.format("%s.yaml", name));
+        return storage.exists(yaml)
+            .thenCompose(
+                exist -> {
+                    final CompletionStage<Boolean> result;
+                    if (exist) {
+                        result = CompletableFuture.completedFuture(true);
+                    } else {
+                        final Key yml = new Key.From(String.format("%s.yml", name));
+                        result = storage.exists(yml);
+                    }
+                    return result;
+                }
+            );
+    }
+
+    /**
+     * Obtains contents from the specified storage.
+     * @return Content of the file.
+     */
+    public CompletableFuture<Content> valueFrom() {
+        throw new UnsupportedOperationException();
+    }
+
+    /**
+     * Trim name of the config file.
+     * @return Filename without extensions if a filename ends with either of
+     *  the two extensions, otherwise empty.
+     */
+    public Optional<String> trimExtension() {
```

Comment:
@genryxy right now it is not really clear what this method returns. Does it trims name and return the extension or does it trim extension or return the name? Method name is `trimExtension` yet comment states "Trim name". This is misleading.

I'd suggest to split the method into two: `isYaml` for filtering files and `name` that returns `String` without extension (or full name if it has no extension). I think it would be easier to understand this way.